### PR TITLE
Ran/assign lesson modal

### PR DESCRIPTION
**Pull request by stormcrow85**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
-  }
+    "name": "infra-dashboard",
+    "version": "v1.0.0",
+    "authors": [
+        "Zinkerz"
+    ],
+    "dependencies": {
+        "angular": "1.4.8",
+        "firebase": "2.4.2",
+        "infra": "https://github.com/Knowledge-OTP/infra.git#v4.0.7",
```
Comment (assafshp) [inline bower.json]: I think it should be 4.0.9

File: dist/znk-infra-dashboard-main.css [hunk 2]
```diff
@@ -1,277 +0,0 @@
-/* fonts size */
```
Comment (assafshp) [inline dist/znk-infra-dashboard-main.css]: why did you remove this file ?

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    'use strict';
+    angular
+        .module('znk.infra-dashboard.assign-lesson')
+        .controller('assignLessonCtrl', function ($scope, $q, locals, SubjectEnum, $translate, $translatePartialLoader, $log, $http, $timeout, TestScoreCategoryEnum) {
+            'ngInject';
+
+            $scope.vm = {};
+            $scope.vm.gridActions = {};
+            $scope.vm.cssClass = locals.cssClass;
+            $scope.vm.modalTitle = 'Assign Lesson';
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: Assign Lesson -> translation

File: src/components/assign-lesson/lessonList.directive.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                                if (!filter(item)) {
+                                    return;
+                                }
+                            }
+                            scope.d.processedData.push(item);
+                        });
+                    }
+
+                    scope.$watch('dataGetter()', function (data) {
+                        console.log('dataGetter() triggered');
```
Comment (assafshp) [inline src/components/assign-lesson/lessonList.directive.js]: remove console.log

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+
+            function _refreshGrid() {
+                if ($scope.vm.gridActions && $scope.vm.gridActions.refresh) {
+                    return $scope.vm.gridActions.refresh();
+                }
+                return $q.reject('refresh grid function not set yet');
+            }
+
+            $scope.vm.submitAssigned = function() {
+                console.log($scope.vm.selectedLessons);
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: $log

Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: doesn't this func should have functionality?

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 6]
```diff
@@ -0,0 +1,315 @@
+/* eslint-disable */
+(function (angular) {
```
Comment (assafshp) [inline src/components/assign-lesson/assign-lesson.controller.js]: who uses this file ? the demo ?

File: src/components/assign-lesson/compile.directive.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('znk.infra-dashboard.assign-lesson');
+
+    module.directive('compile', ['$compile', function ($compile) {
+        return {
```
Comment (assafshp) [inline src/components/assign-lesson/compile.directive.js]: we already have in infra 
angular.module('znk.infra.general').directive('compile'

do we need this one also?

File: src/components/assign-lesson/lessonList.directive.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                        return;
+                    }
+
+                    scope.d = {};
+                    scope.api = {};
+
+                    var getGridApiDefer = $q.defer();
+                    scope.actions.refresh = function () {
+                        getGridApiDefer.resolve();
+                        return getGridApiDefer.promise.then(function () {
```
Comment (assafshp) [inline src/components/assign-lesson/lessonList.directive.js]: what to use getGridApiDefer and not $q.when() ?

File: src/components/assign-lesson/scss/_assignLesson.scss [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+            .clear-filter {
+                margin:0;
+                display: inline-block;
+                cursor: pointer;
+                //svg-icon[name="wrong-icon"] {
+                //    width: 13px;
+                //    height: 13px;
+                //    border: 1px solid $bcColor24;
+                //    border-radius: 50%;
```
Comment (assafshp) [inline src/components/assign-lesson/scss/_assignLesson.scss]: is it needed?

File: src/components/assign-lesson/testScoreCategory.enum.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('TestScoreCategoryEnum.module', []);
+})(angular);
+
+(function (angular) {
+    'use strict';
+
+    angular.module('TestScoreCategoryEnum.module').service('TestScoreCategoryEnum', function(EnumSrv){
```
Comment (assafshp) [inline src/components/assign-lesson/testScoreCategory.enum.js]: is it for ACT ? SAT?

Comment (assafshp) [inline src/components/assign-lesson/testScoreCategory.enum.js]: MAYBE THE service name should be acordinly

File: test/karma-unit.conf.js [hunk 11]
```diff
@@ -22,14 +22,21 @@ module.exports = function (config) {
...
+            'bower_components/angular-translate/angular-translate.js',
+            'bower_components/angular-translate-loader-partial/angular-translate-loader-partial.js',
+            'bower_components/angular-aria/angular-aria.js',
+            'bower_components/angular-messages/angular-messages.js',
+            'bower_components/angular-material/angular-material.js',
+            'bower_components/es5-shim/es5-shim.js',
             'bower_components/angular-mocks/angular-mocks.js',
             'bower_components/mockfirebase/browser/mockfirebase.js',
             'bower_components/jasmine/lib/jasmine-core/jasmine.js',
+            'bower_components/jquery/dist/jquery.js',
```
Comment (assafshp) [inline test/karma-unit.conf.js]: who needs jquery ? what will happen if you remove it

File: src/components/assign-lesson/templates/assignLesson.template.html [hunk 12]
```diff
@@ -0,0 +1,108 @@
+<md-dialog ng-class="vm.cssClass"
+           class="purchase-popup base-border-radius"
+           aria-label="Get Zinkerz"
```
Comment (assafshp) [inline src/components/assign-lesson/templates/assignLesson.template.html]: translation

File: dist/znk-infra-dashboard-main.css (hunk 2 above)
Comment (stormcrow85) [inline dist/znk-infra-dashboard-main.css]: grunt dist removed it and created new files instead

File: bower.json (hunk 1 above)
Comment (stormcrow85) [inline bower.json]: done

File: src/components/assign-lesson/templates/assignLesson.template.html [hunk 13]
```diff
@@ -0,0 +1,108 @@
+<md-dialog ng-class="vm.cssClass"
+           class="purchase-popup base-border-radius"
```
Comment (assafshp) [inline src/components/assign-lesson/templates/assignLesson.template.html]: is it purchase?

File: src/components/assign-lesson/templates/assignLesson.template.html [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+            <div class="search flex-item">
+                <input type="search"
+                       ng-model="vm.searchTerm"
+                       placeholder="Search"
+                       name="search-box"
+                       aria-label="Search">
+                <span class="clear-search"
+                      ng-if="vm.searchTerm"
+                      ng-click="vm.searchTerm = ''">X</span>
```
Comment (assafshp) [inline src/components/assign-lesson/templates/assignLesson.template.html]: what is X?

File: src/components/assign-lesson/templates/assignLesson.template.html [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+                    <md-menu md-offset="-10 60">
+                        <md-button ng-click="$mdOpenMenu($event);"
+                                   aria-label="Filter by subject"
+                                   class="md-icon-button">
+                            <span>{{(vm.currentSubjectName) ? vm.currentSubjectName : vm.subjectLabel }}</span>
+                            <md-icon class="material-icons">{{$mdMenuIsOpen ? 'expand_less': 'expand_more'}}</md-icon>
+                        </md-button>
+                        <md-menu-content class="student-review-grid-filter-menu">
+                            <md-list>
+                                <md-list-item ng-repeat="subject in vm.ACT_Subjects"
```
Comment (assafshp) [inline src/components/assign-lesson/templates/assignLesson.template.html]: ACT_Subjects ? shouldn't it be infra

File: test/karma-unit.conf.js (hunk 11 above)
Comment (stormcrow85) [inline test/karma-unit.conf.js]: done

File: src/components/assign-lesson/assign-lesson.controller.js [hunk 16]
```diff
@@ -0,0 +1,315 @@
...
+    angular
+        .module('znk.infra-dashboard.assign-lesson')
+        .controller('assignLessonCtrl', function ($scope, $q, locals, SubjectEnum, $translate, $translatePartialLoader, $log, $http, $timeout, TestScoreCategoryEnum) {
+            'ngInject';
+
+            $scope.vm = {};
+            $scope.vm.gridActions = {};
+            $scope.vm.cssClass = locals.cssClass;
+            $scope.vm.modalTitle = 'Assign Lesson';
+            $scope.vm.currentStudent = 'Brandon Butler';
```
Comment (stormcrow85) [inline src/components/assign-lesson/assign-lesson.controller.js]: its a demo controller...

File: src/components/assign-lesson/assign-lesson.controller.js (hunk 3 above)
Comment (stormcrow85) [inline src/components/assign-lesson/assign-lesson.controller.js]: its a demo controller...

File: src/components/assign-lesson/assign-lesson.controller.js (hunk 5 above)
Comment (stormcrow85) [inline src/components/assign-lesson/assign-lesson.controller.js]: its a demo controller... it will have once ill integrate with the app

File: src/components/assign-lesson/assign-lesson.controller.js (hunk 6 above)
Comment (stormcrow85) [inline src/components/assign-lesson/assign-lesson.controller.js]: yes

File: src/components/assign-lesson/testScoreCategory.enum.js (hunk 10 above)
Comment (stormcrow85) [inline src/components/assign-lesson/testScoreCategory.enum.js]: its a demo for SAT

File: src/components/assign-lesson/templates/assignLesson.template.html (hunk 13 above)
Comment (stormcrow85) [inline src/components/assign-lesson/templates/assignLesson.template.html]: demo - copy paste

File: src/components/assign-lesson/templates/assignLesson.template.html (hunk 14 above)
Comment (stormcrow85) [inline src/components/assign-lesson/templates/assignLesson.template.html]: Click x to clear current search term

File: src/components/assign-lesson/templates/assignLesson.template.html (hunk 15 above)
Comment (stormcrow85) [inline src/components/assign-lesson/templates/assignLesson.template.html]: demo
